src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactoryHelpers.cs

```diff
@@ -139,47 +136,59 @@ public static bool IsImmutableEnumerableType(this Type type)
             }
         }
 
+        [DynamicDependency("CreateRange`1", "System.Collections.Immutable.ImmutableArray", "System.Collections.Immutable")]
```

Comment:
(nit) can we use the above `const string` fields for these strings? I think it would be easier to track. Same for the assembly name.